wallet/models.py

```diff
@@ -35,15 +35,77 @@ class Wallet(models.Model):
         auto_now_add=True, 
     )
 
+    @property
+    def transactions(self):
+        return (self.transactions_outbound.all() | self.transactions_inbound.all())
 
     def get_absolute_url(self):
         return reverse('api_wallet:detail', kwargs={'pk': self.id})
 
+    def deposit(self, value, transaction):
+        """Deposits a value to the wallet. Also updates the transaction
+        balance_dest field
+        """
+        # Save source values on transaction
+        transaction.balance_dest=self.balance + value
+        transaction.save()
+        
+        # Save values on wallet
+        self.balance += value
+        self.save()
+
+    def withdraw(self, value, transaction):
+        """Withdraw's a value from the wallet.
+        Should the withdrawn amount is greater than the
+        balance this wallet currently has, it raises an
+        :mod:`InsufficientBalance` error. This exception
+        inherits from :mod:`django.db.IntegrityError`. So
+        that it automatically rolls-back during a
+        transaction lifecycle.
+        """
+        if value > self.balance:
+            raise InsufficientBalance('This wallet has insufficient balance.')
+        # Save source values on transaction
+        transaction.wallet_source=self
+        transaction.balance_source = self.balance - value
+        transaction.save()
+        
+        # Save values on wallet
+        self.balance -= value
+        self.save()
+
+    def transfer(self, wallet, value):
```

Comment:
I'd call _destination_wallet_ instead of _wallet_ 

Don't fear to have long names. It's important to have a semantically strong function signature.